feature/topics/src/main/kotlin/uk/govuk/app/topics/ui/TopicScreen.kt

```diff
@@ -36,38 +42,57 @@ internal fun TopicRoute(
     modifier: Modifier = Modifier
 ) {
     val viewModel: TopicViewModel = hiltViewModel()
-    val topic by viewModel.topic.collectAsState()
-
-    TopicScreen(
-        topic = topic,
-        onPageView = { title -> viewModel.onPageView(title) },
-        onBack = onBack,
-        onExternalLink = { section, text, url ->
-            viewModel.onContentClick(
-                section = section,
-                text = text,
-                url = url
-            )
-            onExternalLink(url)
-        },
-        onStepByStepSeeAll = { section, text ->
-            onStepByStepSeeAll()
-            viewModel.onSeeAllClick(
-                section = section,
-                text = text
+    val uiState by viewModel.uiState.collectAsState()
+
+    uiState?.let {
+        when (it) {
+            is TopicUiState.Default -> {
+                it.topicUi?.let { topicUi ->
+                    TopicScreen(
+                        topic = topicUi,
+                        onPageView = { title -> viewModel.onPageView(title) },
+                        onBack = onBack,
+                        onExternalLink = { section, text, url ->
+                            viewModel.onContentClick(
+                                section = section,
+                                text = text,
+                                url = url
+                            )
+                            onExternalLink(url)
+                        },
+                        onStepByStepSeeAll = { section, text ->
+                            onStepByStepSeeAll()
+                            viewModel.onSeeAllClick(
+                                section = section,
+                                text = text
+                            )
+                        },
+                        onSubtopic = { text, ref ->
+                            viewModel.onSubtopicClick(text)
+                            onSubtopic(ref)
+                        },
+                        modifier = modifier
+                    )
+                } ?: run {
+                    ServiceNotRespondingMessage()
+                }
+            }
+
+            is TopicUiState.Offline -> OfflineScreen(
+                topicReference = it.topicReference,
+                onPageView = { title -> viewModel.onPageView(title) },
+                onBack = onBack,
+                onTryAgainClick = { viewModel.getTopic() }
             )
-        },
-        onSubtopic = { text, ref ->
-            viewModel.onSubtopicClick(text)
-            onSubtopic(ref)
-        },
-        modifier = modifier
-    )
+
+            is TopicUiState.ServiceError -> ServiceNotRespondingMessage()
```

Comment:
Incomplete - will be implemented under https://govukverify.atlassian.net/browse/GOVUKAPP-925